File: ruby/j00.arguments.rb
#!/usr/bin/ruby

arg_count   = ARGV.length # get num of args
script_name = $0          # get script name

# check for only 2 arguments
if arg_count != 2 then
   # print helpful instructions 
   puts "\nYou need to enter two numbers: \n\n"
   puts "   Usage: #{script_name} [num1] [num2]\n\n" 
else
   # get sum of both arguments, force cast them to integer
   sum = ARGV[0].to_i + ARGV[1].to_i
   # print results of both arguments (string) and summation (integer)
   puts "The sum of #{ARGV[0]} and #{ARGV[1]} is: #{sum}"
end

File: perl/j00.arguments.pl
#!/usr/bin/perl -w
# acquire num of args and script name
$arg_count   = $#ARGV + 1;    # get num of arguments
$script_name = $0;            # get script name

if ($arg_count != 2) {
  # print helpful instructions
  print "\nYou need to enter two numbers: \n\n";
  print "   Usage: $script_name [num1] [num2]\n\n";
} else {
  $sum = $ARGV[0] + $ARGV[1]; # get sum of both arguments
  # print results of both arguments and summation
  printf  "The sum of %d and %d is: %d.\n", $ARGV[0], $ARGV[1], $sum;
}